xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererDXVA.cpp

```diff
@@ -183,7 +183,9 @@ CRect CRendererDXVA::ApplyTransforms(const CRect& destRect) const
 bool CRendererDXVA::UseToneMapping() const
 {
   // use mapping only if processor doesn't support HDR10
-  return !m_processor->HasHDR10Support() && __super::UseToneMapping();
+  // and is not 10 bit swapchain (HDR to SDR conversion)
+  return !m_processor->HasHDR10Support() && !DX::DeviceResources::Get()->Is10BitSwapchain() &&
```

Comment:
10bit swapchain doesn't mean that tonemapping should be disapbled always.

Reply:
10 bitSwapchain only makes sense if HDR is ON  and if HDR is ON  SW tonemapping should be disabled.

Adaptation of luma levels source HDR to display HDR is done in HW.

Reply:
10bit can be used without HDR. isn't?

Reply:
without HDR why need tonemapping?

4K SDR streams also they look good with 8 bit swapchain

My code assumes 10bitSwapchain = Windows HDR switch ON

However in these conditions (Windows HDR = ON)

SDR Rec709 4K or HD content is rendered properly also.

What is wrong?

Tested with DXVA render:
HDR content  ----> HDR ON  = OK native
SDR content  -----> HDR ON = OK  10 bit swapchain can handle 709 color too
HDR content -----> HDR OFF = OK with HW tonemapping
SDR content ----->  HDR OFF = ON native 

Tested with pixel shaders render:
HDR content  ----> HDR ON  = OK native
SDR content  -----> HDR ON = OK  10 bit swapchain can handle 709 color too
HDR content -----> HDR OFF = OK with SW tonemapping
SDR content ----->  HDR OFF = ON native 

(*)HDR on/off = Windows switch = 10bitswapchain

Reply:
I've spent many hours testing. All these tests are done. Repeated with various sources, local media files, Blu-Ray UHD rip, MKV files (bad-4K-rip) from Netflix and streaming Live TV chanels, some 4K HLG (Hybrid Log-Gamma), 4K SDR, HD and demos source material...
(https: //4kmedia.org/lg-cymatic-jazz-hdr-hlg-uhd-4k-demo/)

Reply:
> My code assumes 10bitSwapchain = Windows HDR switch ON

your code shouldn't assume that. 10bit SC can be used if a user set it in advancedsettings.xml and without HDR capable display. Tonemapping must be enabled for HDR movie and SDR display even if 10bit output is used.

Reply:
who is using try10bitOuput in advancedsettings.xml ?

This is a experimental crap.

What is best? remove experimental crap or discard Windows HDR entire implementation.

Reply:
And by the way, why don't you fix DXVA HW tone mapping in Leia. It's broken and you don't seem to care at all.

Reply:
> who is using try10bitOuput in advancedsettings.xml ?

I'm using it. why not?


> What is best? remove experimental crap or discard Windows HDR entire implementation.

If you named 10bit output by crap doesn't mean that it actually is. Keep both of them.



> And by the way, why don't you fix DXVA HW tone mapping in Leia. It's broken and you don't seem to care at all.

I have no issue with tonemapping at all. How should I guess to fix something?


> Please provide solutions, no problems and broken things.

I'm trying to help you to provide rugth solutions

Reply:
> Keep both of them.

Done